src/api/utils/gasOracle.ts

```diff
@@ -0,0 +1,87 @@
+/*
+  Copyright 2021 Set Labs Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+'use strict';
+
+import axios from 'axios';
+import Assertions from '../../assertions';
+
+import {
+  GasNowData,
+  GasOracleSpeed,
+} from '../../types';
+
+/**
+ * @title GasOracleService
+ * @author Set Protocol
+ *
+ * A utility library for fetching current gas prices by speed for Ethereum and Polygon chains
+ */
+export class GasOracleService {
+  chainId: number;
+  private assert: Assertions;
+
+  static AVERAGE: GasOracleSpeed = 'average';
+  static FAST: GasOracleSpeed = 'fast';
+  static FASTEST: GasOracleSpeed = 'fastest';
+
+  constructor(chainId: number) {
+    this.assert = new Assertions();
+    this.assert.common.isSupportedChainId(chainId);
+    this.chainId = chainId;
+  }
+
+  /**
+   * Returns current gas price estimate for one of 'average', 'fast', 'fastest' speeds.
+   * Default speed is 'fast'
+   *
+   * @param  speed  speed at which tx hopes to be mined / validated by platform
+   * @return        gas price to use
+   */
+  async fetchGasPrice(speed: GasOracleSpeed = 'fast'): Promise<number> {
```

Comment:
Only because ive seen gasnow flake below, did we want to return the ethers default gas price? How does hardhat select the gas price agian?

Reply:
Hardhat defers to ethers which calls the `gasPrice` API of the provider (e.g. Infura, Alchemy, etc)